components-java-sdk/daprdocs/content/en/java-sdk-docs/_index.md

```diff
@@ -0,0 +1,152 @@
+  github_branch: main
+---
+
+Dapr offers packages to help with the development of Java pluggable components.
+
+## Prerequisites
+
+- Java JDK 11 (or greater):
+  - [Oracle JDK](https://www.oracle.com/java/technologies/downloads), or
+  - OpenJDK
+- [Apache Maven](https://maven.apache.org/install.html), version 3.x.
+- [Dapr 1.9 CLI]({{< ref install-dapr-cli.md >}}) or later
+- Initialized [Dapr environment]({{< ref install-dapr-selfhost.md >}})
+- Linux, Mac, or Windows (with WSL)
+
+{{% alert title="Note" color="primary" %}}
+Development of Dapr pluggable components on Windows requires WSL as some development platforms do not fully support Unix Domain Sockets on "native" Windows.
+{{% /alert %}}
+
+## Project creation
+
+Creating a pluggable component starts with an empty Java project.
+
+```bash
+mvn archetype:generate 
+	-DgroupId={project-packaging}
+	-DartifactId={project-name}
+	-DarchetypeArtifactId={maven-template} 
+	-DinteractiveMode=false
+```
+
+## Add packages
+
+Add the Dapr Java pluggable components package to your project.
+
+```bash
+mvn install io.dapr.components
+```
+
+## Create application and service
+
+Creating a Dapr pluggable component application is similar to creating a Java application. Under the standard `src/main/java/io/dapr/components`, `Program.cs`, replace the `WebApplication` related code with the Dapr `PluggableComponent` equivalent.
+
+```java
+package io.dapr.components.server;
+
+public final class PluggableComponent {
+
+  public PluggableComponent(String componentName) {
+    this.name = Objects.requireNonNull(componentName);
+  }
+
+  public static PluggableComponent withName(String componentName) {
+    return new PluggableComponent(componentName);
+  }
+
+  // Register a new {@link StateStore} with this pluggable component.
+
+  public PluggableComponent withStateStore(StateStore stateStore) {
+    Objects.requireNonNull(stateStore);
+
+    exposedServices.add(new StateStoreGrpcComponentWrapper(stateStore)); 
+
+    return this;
+  }
+
+  // Repeat for each component 
+  
+}
+```
+
+This creates an application with a single service. Each service:
+
+- Corresponds to a single Unix Domain Socket
+- Can host one or more component types
+
+{{% alert title="Note" color="primary" %}}
+Only a single component of each type can be registered with an individual service. However, [multiple components of the same type can be spread across multiple services]({{< ref dotnet-multiple-services >}}).
+{{% /alert %}}
+
+## Implement and register components
+
+ - [Implementing an input/output binding component]({{< ref java-bindings >}})
+ - [Implementing a pub-sub component]({{< ref java-pub-sub >}})
+ - [Implementing a state store component]({{< ref java-state-store >}})
+
+## Test components locally
+
+Pluggable components can be tested by starting the application on the command line and configuring a Dapr sidecar to use it.
+
+To start the component, in the application directory:
+
+```bash
+mvn 
```

Comment:
It has been ages but I don't think that by default `mvn` would start the component.